### DISCUSSION TITLE
Glue library with FluentValidation

### DISCUSSION THREAD

**Discussion by Kralizek** [AutoFixture/AutoFixture#1461] [Q&A]
Maybe more of an idea than an actual Q&A but I ended up on a project using FluentValidation to validate incoming HTTP payloads and I'd love not to replicate the validation rules on the model just for AF advantage.

I stumbled upon this post on SO and I wonder if this is still the case: https://stackoverflow.com/questions/68740987/autofixture-adhering-to-fluentvalidation.

**Comment by aivascu** [comment]
I remember playing with the idea at the time. I was able to get the tree of rules from a validator object. 
I think I even started translating that to a tree of specifications and builders. Then I realized there is the predicate validator and it is possible to inject other validators and other services in a validator.
I kinda gave up at that point. I should revisit it, and publish the WIP, maybe someone will be able to expand on it/make a proper extension.

**Reply by Kralizek** [reply]
I think it's understandable if not all the rules can be converted into specifications and builders.

I don't think all Data Annotations are supported either 🤔

**Reply by aivascu** [reply]
I think data annotations integration was able to get away with not adopting all attributes because of its design limitation.
You can't inject services in attributes, and writing custom data annotation attributes is not very convenient, and not commonly practiced.

However... even with the limitations of data annotations people got creative, and use them in non-standard ways.
For instance #1409 requests that AutoFixture generates a `enum` value for a `string` member that is decorated with `EnumDataTypeAttribute`. There are other similar issues opened.

I mean, I could try to make a glue package for fluent validation, but it would only be able to support very basic validators. AutoFixture would generate objects that are close but not 100% matching the validator, thus failing validation.

I'm concerned the library would be constantly spammed with issues requesting support for a specific scenarios.

**Reply by Kralizek** [reply]
> I'm concerned the library would be constantly spammed with issues requesting support for a specific scenarios.

That is a valid point :(
